editor/plugins/version_control_editor_plugin.cpp

```diff
@@ -133,149 +140,278 @@ void VersionControlEditorPlugin::_initialize_vcs() {
 
 	ERR_FAIL_COND_MSG(!EditorVCSInterface::get_singleton()->initialize(res_dir), "VCS was not initialized");
 
+	if (set_up_username->get_text() == "" || set_up_password->get_text() == "") {
+		WARN_PRINT("Some feature will not work without remote credentials.")
+	}
+
+	EditorVCSInterface::get_singleton()->set_up_credentials(set_up_username->get_text(), set_up_password->get_text());
 	_refresh_stage_area();
+	_refresh_commit_list();
+	_refresh_branch_list();
 }
 
-void VersionControlEditorPlugin::_send_commit_msg() {
+void VersionControlEditorPlugin::_refresh_branch_list() {
+	Array branch_list = EditorVCSInterface::get_singleton()->get_branch_list();
+	branch_select->clear();
+	for (int i = 0; i < branch_list.size(); i++) {
+		branch_select->add_item(branch_list[i], i);
+	}
 
-	String msg = commit_message->get_text();
-	if (msg == "") {
+	// TODO: Add new brach feature
+	// branch_select->add_separator();
+	// branch_select->add_item("New Branch");
+	branch_select->select(0); // First branch in the list is current branch
+}
+
+void VersionControlEditorPlugin::_refresh_commit_list() {
 
-		commit_status->set_text(TTR("No commit message was provided"));
+	if (!EditorVCSInterface::get_singleton()) {
+		WARN_PRINT("No VCS plugin is initialized. Select a Version Control Addon from Project menu");
 		return;
 	}
 
-	if (EditorVCSInterface::get_singleton()) {
+	commit_list->get_root()->clear_children();
 
-		if (staged_files_count == 0) {
+	Array commits_info_list = EditorVCSInterface::get_singleton()->get_previous_commits();
+	for (int i = 0; i < commits_info_list.size(); i++) {
+		Dictionary commit_info = commits_info_list[i];
+		TreeItem *item = commit_list->create_item(commit_list->get_root());
+		item->set_text(0, String(commit_info["message"]).strip_edges());
+		item->set_text(1, String(commit_info["author"]).strip_edges());
+	}
+}
 
-			commit_status->set_text(TTR("No files added to stage"));
-			return;
-		}
+void VersionControlEditorPlugin::_commit() {
+	String msg = commit_message->get_text();
 
-		EditorVCSInterface::get_singleton()->commit(msg);
+	if (msg == "") {
+		WARN_PRINT(TTR("No commit message was provided"));
+		return;
+	}
 
-		commit_message->set_text("");
-		version_control_dock_button->set_pressed(false);
-	} else {
+	if (!EditorVCSInterface::get_singleton()) {
+		WARN_PRINT("No VCS plugin is initialized. Select a Version Control Addon from Project menu");
```

Comment:
The two next tests should like use ERR_FAIL_COND_V_MSG instead. This is usually what we do when a function fails.
I think we should consider this as an error (and not a warning) because the commit button should be disabled when the two following conditions are not met.